### ISSUE TITLE
"Unexpected low cardinality version" error when performing a query including LowCardinality(String) columns in GROUP BY

### ISSUE THREAD

**Issue by georgipeev** [ClickHouse/clickhouse-connect#232]
### Describe the bug
When a somewhat complex query with a lot of aggregation is performed via `query_df`, `clickhouse_connect` fails with the following error:
```
clickhouse_connect.datatypes.base:Unexpected low cardinality version 7450489120435229448 reading type LowCardinality(String)
ERROR:clickhouse_connect.datatypes.registry:Unrecognized ClickHouse type base: ssMarketValueDecimal name: ssMarketValueDecimal(30, 12)NetMarketValueDecimal(30, 12)GrossPLDecimal(30, 12)NetPLDecimal(30, 12)Tota
```

### Steps to reproduce
The query in question:
```
SELECT TradeDate, ProductGroup, Strategy, Sector, 
argMax(LongMarketValue, TimeSlice) AS LongMarketValue, 
argMax(ShortMarketValue, TimeSlice) AS ShortMarketValue, 
argMax(GrossMarketValue, TimeSlice) AS GrossMarketValue, 
argMax(NetMarketValue, TimeSlice) AS NetMarketValue, 
argMax(GrossPL, TimeSlice) AS GrossPL, 
argMax(NetPL, TimeSlice) AS NetPL, 
argMax(TotalPosMax, TimeSlice) AS TotalPosMax, 
argMax(TotalPosMin, TimeSlice) AS TotalPosMin, 
sum(MarketVolume) AS MarketVolume
FROM pnl
WHERE TradeDate BETWEEN '2023-08-20' AND '2023-08-22' 
AND Namespace = 'foo'
GROUP BY TradeDate, ProductGroup, Strategy, Sector
```

Of note is that the group-by columns `ProductGroup`, `Strategy`, and `Sector` are all of type `LowCardinality(String)`.

The underlying data is proprietary and, even if obfuscated, is far too large to be included.
The above query is trimmed down to be minimal and still exhibit the problem. Modifying it slightly causes the problem to no longer appear. Tested changes that yield a correctly working query:
- Remove one of the `argMax` fields
- Remove one of the `sum` fields
- Reduce the `TradeDate`  range (increasing it causes the problem to persist)
- Remove the `Namespace` condition (problem disappears despite more rows being selected)

Additionally, performing the query via the `clickhouse-client` command-line tool works fine, as does doing it via `query_arrow` with `output_format_arrow_low_cardinality_as_dictionary=0`. I believe the error implies a problem in `query_df` with reading the contents of a `LowCardinality(String)` column as a categorical type, which `query_arrow` does not exhibit probably because I explicitly ask it to not use `pyarrow.DictionaryType`.

### clickhouse-connect and/or ClickHouse server logs

Nothing of note.

#### Environment
* clickhouse-connect version: 0.5.20
* Python version: 3.9.7
* Operating system: RHEL 8.6

#### ClickHouse server
* ClickHouse Server version: 22.8.16 revision 54460

**Comment by genzgd** [comment]
I believe this is a duplicate of https://github.com/ClickHouse/clickhouse-connect/issues/221.  Can you check if it's fixed by either upgrading ClickHouse to 22.9+ or clickhouse-connect to 0.6.8+

**Comment by georgipeev** [comment]
Upgrading `clickhouse-connect` to 0.6.9 solves the issue. As usual, thanks for the prompt response!
